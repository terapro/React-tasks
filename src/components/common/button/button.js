import React, {Component} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export class Button extends Component {
  constructor(props) {
    super(props);

    this.callbackFunc = () => {
      this.props.btnClick(this.props.content);
    };
  }
  render() {
    const {btnSm, btnBig, btnWhite, btnRed, btnGrey, btnBold, btnTransp, btnTranspRed, btnUC} = this.props;
    const {id, content} = this.props;
    let itemClass = classNames({
      'btn': true,
      'btn-sm': btnSm,
      'btn-big': btnBig,
      'btn-white': btnWhite,
      'btn-red': btnRed,
      'btn-grey': btnGrey,
      'btn-bold':  btnBold,
      'btn-transp': btnTransp,
      'btn-transp-red': btnTranspRed,
      'btn-UC': btnUC
    });
    return( <div className={itemClass} onClick = {this.callbackFunc} id={id}>
      {content}
    </div>);
  }
};


Button.propTypes = {
  btnSm: PropTypes.bool,
  btnBig: PropTypes.bool,
  btnWhite: PropTypes.bool,
  btnRed: PropTypes.bool,
  btnGrey: PropTypes.bool,
  btnBold: PropTypes.bool,
  btnTransp: PropTypes.bool,
  btnTranspRed: PropTypes.bool,
  btnUC: PropTypes.bool,
  content: PropTypes.string,
  btnClick: PropTypes.func.isRequired,
  id: PropTypes.string
};

Button.defaultProps = {
  btnSm: false,
  btnBig: false,
  btnWhite: false,
  btnRed: false,
  btnGrey: false,
  btnBold: false,
  btnTransp: false,
  btnTranspRed: false,
  btnUC: false,
  content: 'Button',
  id: ''
};