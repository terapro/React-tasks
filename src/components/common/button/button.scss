.btn {
  height: $elementHeight;
  line-height: $elementHeight;
  text-align: center;
  padding: 0 $elementPadding;
  font-size: $elementFontSize;
  letter-spacing: 1px;
  cursor: pointer;
  border-radius: 5px;
  font-weight: normal;
  user-select: none;
}

.btn:hover {

}

.btn-sm {
}

.btn-big {
  font-size: 18px;
  line-height: 45px;
  height: 45px;
  padding: 0 25px;
  min-width: 235px;
}

.btn-white {
  background: $color_white;
  color: $color_red;
}

.btn-red {
  background: $color_red;
  color: $color_white;
}

.btn-grey {
  background: $color_grey;
  color: $color_white;
}
.btn-transp {
  color: $color_black;
}
.btn-transp-red {
  color: $color_red;
}

.btn-bold {
  font-weight: bold;
}
.btn-UC {
  text-transform: uppercase;
}



