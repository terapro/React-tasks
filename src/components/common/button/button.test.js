import React from 'react';
import {Button} from 'src/components/common/button/button.js';
import {shallow} from 'enzyme';


const mockFn = jest.fn();
const mockBtnClass= 'btn';

describe('<Button />', () => {
  it('should render the button', () => {
    const wrapper = shallow(<Button btnClick={mockFn} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('invokes onClick', () => {
    shallow(<Button btnClick={mockFn} />)
      .find('div' + '.' + mockBtnClass)
      .simulate('click');
    expect(mockFn).toBeCalled();
  });
  it('should be configurable', () => {
    expect(shallow(<Button btnRed btnClick={mockFn}/>).is('.btn-red')).toBe(true);
    expect(shallow(<Button btnTransp btnClick={mockFn}/>).is('.btn-transp')).toBe(true);
  });

});












// Unit test for react

/*import React from 'react';
import ReactDOM from 'react-dom';
import {Button} from './button.js';*/




/*
describe('<Button>', () => {
  test('', () => {

  });

  describe('methods', () => {

  })
})

test('mounts', () => {
  const wrapper = shallow(<Button />);

  expect(wrapper).toMatchSnapshot();

  const div = document.createElement('div');
  ReactDOM.render(<Button btnClick = {() => {}} />, div);

})*/
