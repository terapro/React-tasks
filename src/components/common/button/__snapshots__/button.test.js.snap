// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
<div
  className="btn"
  id=""
  onClick={[MockFunction]}
>
  Button
</div>
`;

exports[`Testing Button render should render the button 1`] = `
<div
  className="btn"
  id=""
  onClick={[MockFunction]}
>
  Button
</div>
`;
